src/hooks/useGeolocation.js

```diff
@@ -0,0 +1,11 @@
+import { useSelector } from 'react-redux';
+
+/**
+ * Hook to track the geolocation status in the browser.
+ *
+ * @returns {boolean} Geolocation enabled flag.
+ */
+export default function useGeolocationEnabled() {
+  const enabled = useSelector(state => state.userData.locationEnabled);
```

Comment:
I think it will be more valuable to convert this to a selector function rather than a hook that explicitly uses `useSelector`

Reply:
That's fair, I had initially planned to put a ton of logic inside the hook, but then moved away from that idea.